src/sas/qtgui/MainWindow/DataExplorer.py

```diff
@@ -726,14 +726,13 @@ def isItemReady(index):
 
         if len(selected_items) < 1:
             return
-
-        # Which perspective has been selected?
-        if len(selected_items) > 1 and not self._perspective().allowBatch():
+        #Check that you have only one box item checked when swaping data
+        if len(selected_items) > 1 and (self.chkSwap.isChecked() or not self._perspective().allowBatch()):
             if hasattr(self._perspective(), 'title'):
                 title = self._perspective().title()
             else:
                 title = self._perspective().windowTitle()
-            msg = title + " does not allow multiple data."
+            msg = title + " does not allow replacing multiple data."
```

Comment:
You probably want to use the perspective name rather than the `title()`  method, since `title()` returns the full string including the fit algorithm, so we get a pretty confusing message
`Fit panel - Active Fitting Optimizer: Levenberg-Marquardt does not allow replacing multiple data`.